File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptSetOptStringValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.test.TypeTester;
import com.dslplatform.ocd.values.OptSetOptStringInValue.OptSetOptStringValue;
import com.dslplatform.patterns.ServiceLocator;
import java.util.*;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptSetOptStringValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
        locator = Bootstrap.init(TestOptSetOptStringValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertTrue(TypeTester.testField(OptSetOptStringValue.class, "optSetOptString")
                .resultEquals(Set.class, String.class));
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertTrue(TypeTester.testGetter(OptSetOptStringValue.class, "getOptSetOptString")
                .resultEquals(Set.class, String.class));
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertTrue(TypeTester.testSetter(OptSetOptStringValue.class, "setOptSetOptString", Set.class, String.class)
                .resultEquals(OptSetOptStringValue.class));
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptSetOptStringValue.scala
package com.dslplatform.ocd.values

import OptSetOptStringInValue.OptSetOptStringValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptSetOptStringValue extends SpecExtensions {
  def `Test property field type: Option[Set[Option[String]]]`() =
    checkType(
      typeOf[Option[Set[Option[String]]]]
    , typeOf[OptSetOptStringValue].member("optSetOptString": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptSetOptStringValue().optSetOptString === None)
  }
}
